src/MSBuild/XMake.cs

```diff
@@ -620,6 +622,14 @@ string commandLine
                         Environment.SetEnvironmentVariable("MSBUILDLOADALLFILESASWRITEABLE", "1");
                     }
 
+                    // Honor the low priority flag, we place our selves below normal
+                    // priority and let sub processes inherit that priority.
+                    if (lowPriority)
+                    {
+                        Process currentProc = Process.GetCurrentProcess();
```

Comment:
I'd move all these priority changes to a new utility class, with commands like `PriorityUtils.SwitchToUserLowPriority` that encapsulate the meaning of a "user wants to do other things with their computer" request and how to achieve it. Another use future potential use case for the util class would be encapsulating the meaning of "node spawned and needs high priority to respond back"